src/vs/workbench/api/node/extHost.protocol.ts

```diff
@@ -257,6 +264,12 @@ export abstract class ExtHostEditorsShape {
 	$acceptTextEditorRemove(id: string): void { throw ni(); }
 }
 
+export abstract class ExtHostTreeExplorersShape {
```

Comment:
`InternalTreeExplorerNode` doesn't have any method. It's the way we communicate between `ExtHost/MainThread`. Node will be provided and stored at `ExtHost` side. For each node we use `getLabel` etc to get info from the node, and construct a slim internal node to send to `MainThread`.